transaction/confidential.go

```diff
@@ -0,0 +1,66 @@
+package transaction
```

Comment:
This should not stay in the `transaction` package, but rather we should create a dedicated `confidential` one.

Reply:
done